src/nvim/api/private/helpers.c

```diff
@@ -758,6 +757,44 @@ bool buf_collect_lines(buf_T *buf, size_t n, int64_t start, bool replace_nl, Arr
   return true;
 }
 
+/// Returns a substring of a buffer line
+String buf_get_line(buf_T *buf, int64_t lnum, int64_t start_col, int64_t end_col, bool replace_nl,
```

Comment:
Minor nit, but it's a little confusing to have buf_get_line here have different semantics than nvim_buf_get_lines since this is slicing a line and the latter is retrieving lines.

Reply:
I’ll rename to `buf_get_partial_line` or `buf_get_line_substr` or something

Reply:
I went with `buf_get_text`, since it's a helper for `nvim_buf_get_text` and is the "meat" of the implementation. I'm not married to this name though so if anyone has a better suggestion I'm open to it.